src/components/Header.tsx

```diff
@@ -1,33 +1,42 @@
 import React, { Dispatch, SetStateAction } from 'react';
-// @ts-ignore
+
 import logo from '../image/logo.png';
 
 interface IHeader {
-  setSignInModalOn: Dispatch<SetStateAction<boolean>>;
-  setSignUpModalOn: Dispatch<SetStateAction<boolean>>;
+  setSetTitleModal: Dispatch<SetStateAction<string>>;
+  setIsOpen: Dispatch<SetStateAction<boolean>>;
 }
 
-export function Header({ setSignInModalOn, setSignUpModalOn }: IHeader) {
-  const onClickSignInHandler = () => {
-    setSignInModalOn(true);
-  };
-  const onClickSignUpHandler = () => {
-    setSignUpModalOn(true);
+export function Header({ setSetTitleModal, setIsOpen }: IHeader) {
+  const onClickHandler = (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+    setSetTitleModal(e.currentTarget.textContent!);
+    setIsOpen(true);
   };
+
   return (
-    <header className="h-26 w-4/5 m-auto">
-      <div className="flex justify-between items-center">
-        <img width={80} src={logo} alt="main-logo" />
-        <div className="w-36 flex justify-between">
-          <div className="text-green-300 hover:text-green-700" onClick={onClickSignInHandler}>
-            Sign In
-          </div>
-          <div className="text-green-300 hover:text-green-700" onClick={onClickSignUpHandler}>
-            Sign Up
+    <header className="h-28 w-4/5 m-auto">
+      <div className="h-full flex items-center">
+        <div className="w-full flex justify-between items-center">
+          <a href="/">
+            <img width={80} src={logo} alt="main-logo" />
+          </a>
+          <div className="w-36 flex justify-between">
+            <div
+              className="text-green-300 hover:text-green-700 cursor-pointer"
+              onClick={onClickHandler}
```

Comment:
The idea of getting the text content this way and making the logic of which form to show dependent on this doesn't seem a solid solution. What if the text changes in the future . or i18n is added.